src/components/Calendar.ts

```diff
@@ -0,0 +1,176 @@
+import { Component, ReactChild, createElement } from "react";
+
+import { Alert } from "./Alert";
+import * as BigCalendar from "react-big-calendar";
+import { CalendarLoading } from "./CalendarLoading";
+import { DragDropContext } from "react-dnd";
+import HTML5Backend from "react-dnd-html5-backend";
+import * as globalize from "globalize";
+import localizer from "react-big-calendar/lib/localizers/globalize";
+import * as withDragAndDrop from "react-big-calendar/lib/addons/dragAndDrop";
+
+import "react-big-calendar/lib/addons/dragAndDrop/styles.css";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import "../ui/Calendar.scss";
+
+localizer(globalize);
+const DragAndDropCalendar = withDragAndDrop(BigCalendar);
+
+export interface CalendarProps {
+    alertMessage: ReactChild;
+    events: CalendarEvent[];
+    color: string;
+    loading?: boolean;
+    showMultiDayTimes?: boolean;
+    defaultView: View;
+    startPosition: Date;
+    firstDay?: number;
+    messages?: CustomViews[];
+    popup: boolean;
+    resizable: boolean;
+    selectable: boolean;
+    dayFormat?: string;
+    weekdayFormat?: string;
+    timeGutterFormat?: string;
+    monthHeaderFormat?: string;
+    dayHeaderFormat?: string;
+    views?: string;
+    onSelectEventAction?: (eventInfo: object) => void;
+    onEventResizeAction?: (eventInfo: any) => void;
+    onSelectSlotAction?: (slotInfo: object) => void;
+    onEventDropAction?: (eventInfo: object) => void;
+}
+
+export type View = "month" | "week" | "work_week" | "day" | "agenda";
+
+export interface CalendarEvent {
+    title: string;
+    allDay: boolean;
+    start: Date;
+    end: Date;
+    guid: string;
+    color: string;
+}
+
+export interface CustomViews {
+    customCaption: string;
+    customView: View;
+}
+
+interface CalendarState {
+    events?: CalendarEvent[];
+    color?: string;
+}
+
+class Calendar extends Component<CalendarProps, CalendarState> {
+    state: CalendarState = { events: this.props.events };
+
+    componentWillMount() {
+        globalize().cultures.default.calendars.standard.firstDay = window.mx.session.sessionData.locale.firstDayOfWeek;
+    }
+
+    render() {
+        if (this.props.alertMessage) {
+            return createElement(Alert, { className: "widget-calendar-alert" }, this.props.alertMessage);
+        }
+        if (this.props.loading) {
+            return createElement(CalendarLoading);
```

Comment:
when is loading showing.. what is it function? Can we just show the calendar rendered, with an additional class, overloaying with loading indicator in the same way as we do with the LVC